### ISSUE TITLE
Could anyone pls make a simple guide on how to build espeak-ng on windows?

### ISSUE THREAD

**Issue by vcjob** [espeak-ng/espeak-ng#807]
Hi everyone!

I am having a bad time trying to build espeak-ng on windows. 
First of all, pls. do not advice to use "ready to use .exe and .msi".

By a simple guide I mean some script like "First download and install MS Visual Studio 2017 Community edition. Then Install Wix... Then find msbuild AT FOLDER (folder, as there are a lot of them), invoke msbuild /p (some paramters)... espeak-ng.sln" and so on.
The thing is, I gave up resolving errors like `stddef.h: No such file or directory` `ctype.h: No such file or directory` `errno.h: No such file or directory` `vcbuild.exe not found` and so on, installing more and more adds-on to Visual Studio, which is already more than 12 GB!
I assume I go the wrong way, and there must be something I missed. That's so easy to build it on the linux systems, why it's so painful on Windows... God!

**Comment by Lamashino** [comment]
Yes, you are right. I had similar issues and I "googled" solutions for those problems with Visual Studio 2019.

In particular, to fix the "stddef.h" and some others that will arise, you have to add to the project properties the path to the "includes" (in my case: C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt) and the path to the libraries (C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\ucrt\x64) into the Libraries section. 

In addition to this, you will probably have problems with some definitions as "IS_ISDIR" and similar. I just added them at the beginning of the files (only in my "personal" copy of the project).

**Comment by rhdunn** [comment]
@Lamashino The `S_ISDIR` and other definitions should be handled by `src/include/compat/sys/stat.h`, etc. It may be an ordering issue given that you added include directories to the path -- you may need to move those to the end of the list so it checks them last. That is, so the compat file is included first, then that includes the Microsoft header.

**Comment by vcjob** [comment]
@Lamashino , @rhdunn Thank you for the answers! 
I fixed multiple error, including "stddef.h", M_PI, S_ISDIR... Now fighting with the `Error LNK2019: int __cdecl invoke_main(void) ( invoke_main@@yahxz)` and `LNK1120`. Strange, since I've got my linker-system-subsystem set to _Windows (/SUBSYSTEM:WINDOWS)_

**Comment by rojanu** [comment]
I've been trying to compile, tried above suggestions to no avail. is there guide/steps available anywwhere or could some one post them here?

**Comment by YousafKhan1** [comment]
@vcjob 
Please I have the same problem if you explain it for me how to solve this problem with full detail.
